tests/rules/test_result.py

```diff
@@ -72,14 +71,21 @@ def test_result_err_keys(messages, true_err_keys):
                     {"s": [0.25]}, index=["us"], name="Coverage for boolean fields"
                 )
             ],
-            ["<h4>test show</h4>", "summary"],
         ),
     ],
 )
-def test_show(mocker, capsys, message, stats, outputs):
-    mock_pio_show = mocker.patch("plotly.io.show", autospec=True)
-    mocked_md = mocker.patch("arche.report.display_markdown", autospec=True)
+def test_show(mocker, capsys, message, stats):
+    mocked_display = mocker.patch("arche.report.display_html", autospec=True)
     res = create_result("test show", message, stats=stats)
     res.show()
-    mock_pio_show.assert_called_once_with(res.figures[0])
-    mocked_md.assert_has_calls(mocker.call(o) for o in outputs)
+    report_html = get_report_from_iframe(mocked_display.mock_calls[0][1][0])
+    assert report_html.count("Plotly.newPlot") == 1
+
+
+def test_outcome_equality():
+    outcomes = [Outcome.SKIPPED, Outcome.PASSED, Outcome.FAILED, Outcome.WARNING]
```

Comment:
> Good, but the simplest way:
> 
> ```
> assert Outcome.SKIPPED == Outcome.SKIPPED
> assert Outcome.WARNING != Outcome.PASSED
> ```
> 
> I love the simplest solution which does the job because it's easier to read. It's up to you if you want to change the tests or not.

Yes, but doing so wouldn't the code need to have several lines in order to test every possible pair of values?

Reply:
We don't have to test every possible pair. The goal is to test `__eq__ and __ne__` operators, so what can go wrong:
1. anyone of them is not implemented
2. anyone of them is implemented wrong
Thus, if we have at least one use case to test both operators, they work. I.e. if there's `Outcome.SKIPPED == Outcome.SKIPPED` we know all others work (since they all belong to the single class`, same with `!=`.
Makes any sense?

When I write a unit test, I am thinking what can go wrong from human side (where the developer might make an error). It's either the function doesn't return what's expected or doesn't cover all use cases.